packages/base-types/src/node/carousel.ts

```diff
@@ -42,14 +36,14 @@ export interface StepPorts extends BaseStepPorts<StepBuiltInPorts, []> {}
 
 export interface StepData extends BaseNoMatchStepData, BaseNoReplyStepData {
   layout: CarouselLayout;
-  cards: CarouselCard[];
+  cards: CarouselCard<GeneralRequestButton | ActionRequestButton>[];
 }
 
 export interface Step<Data = StepData> extends BaseStep<Data, StepPorts> {
   type: NodeType.CAROUSEL;
 }
 
-export type NodeCarouselCard = CarouselCard<GeneralRequestButton>;
+export type NodeCarouselCard = CarouselCard<GeneralRequestButton | ActionRequestButton>;
```

Comment:
only this should be changed, let's add a new line after, just to improve readability
```suggestion
export type NodeCarouselCard = CarouselCard<GeneralRequestButton | ActionRequestButton>;

```